src/frontends/pytorch/src/translate_session.cpp

```diff
@@ -257,6 +281,111 @@ size_t TranslateSession::decode_tensor_name(const Output<Node>& output) {
+    auto to_be_inserted_data_1d = std::make_shared<opset10::Reshape>(value, neg_one_1d, false);
+    auto updated_data_1d =
+        std::make_shared<opset10::ScatterNDUpdate>(to_insert_data_1d, data_indices_1d, to_be_inserted_data_1d);
+
+    // Reshape to initial shape
+    return std::make_shared<opset10::Reshape>(updated_data_1d, input_shape, false);
+}
+
+Output<Node> select_backprop(std::shared_ptr<TorchDecoder> node, Output<Node> select_output, Output<Node> value) {
+    auto gather_node = select_output.get_node_shared_ptr();
+    FRONT_END_OP_CONVERSION_CHECK(ov::as_type_ptr<opset10::Gather>(gather_node),
+                                  "Conversion rule for aten::select doesn't contain Gather node.");
+
+    auto zero = opset10::Constant::create(element::i64, Shape{}, {0});
+    auto one = opset10::Constant::create(element::i64, Shape{}, {1});
+    auto neg_one_1d = opset10::Constant::create(element::i64, Shape{1}, {-1});
+    auto scattering_shape = opset10::Constant::create(element::i64, Shape{2}, {-1, 1});
+
+    // Get 1d indices [0..numel)
+    auto to_insert_data = gather_node->input_value(0);
+    auto input_shape = std::make_shared<opset10::ShapeOf>(to_insert_data, element::i64);
+    auto numel = std::make_shared<opset10::ReduceProd>(input_shape, zero, false);
+    auto full_data_indices_1d = std::make_shared<opset10::Range>(zero, numel, one, element::i64);
+
+    // Slice indices by same start, stop, slice, axes as initial Slice
+    auto full_data_indices = std::make_shared<opset10::Reshape>(full_data_indices_1d, input_shape, false);
+    Output<Node> data_indices =
+        std::make_shared<opset10::Gather>(full_data_indices, gather_node->input_value(1), gather_node->input_value(2));
+
+    // Scatter in flattened tensor with indices and flattened data to be inserted
+    auto to_insert_data_1d = std::make_shared<opset10::Reshape>(to_insert_data, neg_one_1d, false);
+    auto data_indices_1d = std::make_shared<opset10::Reshape>(data_indices, scattering_shape, false);
+    auto to_be_inserted_data_1d = std::make_shared<opset10::Reshape>(value, neg_one_1d, false);
+    auto updated_data_1d =
+        std::make_shared<opset10::ScatterNDUpdate>(to_insert_data_1d, data_indices_1d, to_be_inserted_data_1d);
+
+    // Reshape to initial shape
+    return std::make_shared<opset10::Reshape>(updated_data_1d, input_shape, false);
+}
+}  // namespace
+
+using BackpropCreatorFunction =
+    std::function<ov::Output<ov::Node>(std::shared_ptr<TorchDecoder>, Output<Node>, Output<Node>)>;
+
+Output<Node> TranslateSession::get_backprop_op(std::shared_ptr<TorchDecoder> node,
+                                               Output<Node> direct_op_output,
+                                               Output<Node> value) {
+    std::map<std::string, BackpropCreatorFunction> backprop_map = {
+        {"aten::slice", slice_backprop},
+        {"aten::select", select_backprop},
+    };
```

Comment:
not clear why is this map needed? Should we create for other operations? Also, does it make sense to place them into operation specific files?